camelcase_test.go

```diff
@@ -1,35 +1,47 @@
 package camelcase
 
-import (
-	"reflect"
-	"testing"
-)
+import "fmt"
 
-func TestSplit(t *testing.T) {
-	var testCases = []struct {
-		input  string
-		output []string
-	}{
-		{input: "", output: []string{}},
-		{input: "lowercase", output: []string{"lowercase"}},
-		{input: "Class", output: []string{"Class"}},
-		{input: "MyClass", output: []string{"My", "Class"}},
-		{input: "MyC", output: []string{"My", "C"}},
-		{input: "HTML", output: []string{"HTML"}},
-		{input: "PDFLoader", output: []string{"PDF", "Loader"}},
-		{input: "AString", output: []string{"A", "String"}},
-		{input: "SimpleXMLParser", output: []string{"Simple", "XML", "Parser"}},
-		{input: "vimRPCPlugin", output: []string{"vim", "RPC", "Plugin"}},
-		{input: "GL11Version", output: []string{"GL", "11", "Version"}},
-		{input: "99Bottles", output: []string{"99", "Bottles"}},
-		{input: "May5", output: []string{"May", "5"}},
-		{input: "BFG9000", output: []string{"BFG", "9000"}},
-	}
+func ExampleSplit() {
 
-	for _, c := range testCases {
-		res := Split(c.input)
-		if !reflect.DeepEqual(res, c.output) {
-			t.Errorf("input: '%s'\n\twant: %v\n\tgot : %v\n", c.input, c.output, res)
-		}
+	for _, c := range []string{
+		"",
+		"lowercase",
+		"Class",
+		"MyClass",
+		"MyC",
+		"HTML",
+		"PDFLoader",
+		"AString",
+		"SimpleXMLParser",
+		"vimRPCPlugin",
+		"GL11Version",
+		"99Bottles",
+		"May5",
+		"BFG9000",
+		"BöseÜberraschung",
+		"Two  spaces",
+		"BadUTF8\xe2\xe2\xa1",
+	} {
+		fmt.Printf("%#v\n", Split(c))
 	}
+
+	// Output:
+	// []string{}
+	// []string{"lowercase"}
+	// []string{"Class"}
+	// []string{"My", "Class"}
+	// []string{"My", "C"}
+	// []string{"HTML"}
+	// []string{"PDF", "Loader"}
+	// []string{"A", "String"}
+	// []string{"Simple", "XML", "Parser"}
+	// []string{"vim", "RPC", "Plugin"}
+	// []string{"GL", "11", "Version"}
+	// []string{"99", "Bottles"}
+	// []string{"May", "5"}
+	// []string{"BFG", "9000"}
+	// []string{"Böse", "Überraschung"}
+	// []string{"Two", "  ", "spaces"}
+	// []string{"BadUTF8\xe2\xe2\xa1"}
```

Comment:
@fatih This new example test covers the same test cases, plus a couple more. Can you confirm if you'd like to have the previous test reinstated because it was easier to visualise the string and the split result on the same line, or if it was just that it wasn't clear that the new test covers all of the original test cases?

My preferred solution to get the results on the same line would be that I'd update the example test not to just output the splits, but also the original string, so we'd have something like:

```
// Output:
// "lowercase" => []string{"lowercase"}
// "Class" => []string{"Class"}
// "MyClass" => []string{"My", "Class"}
```

The advantages of this approach would be:

* The string, and the splitting results appear on the same line, making easier to edit
* Tests are not duplicated as regular tests and example tests
* The behaviour of the library is clearer to see when reading the examples

Would this also be ok with you?

Maybe this is